### ISSUE TITLE
EZ4 with no DS mode

### ISSUE THREAD

**Issue by ikorobus** [AdmiralCurtiss/nds-savegame-manager#2]
Hello, sorry if this comes off as redundant or stupid question but I just can't seem to make this work on my EZ4 with the latest 2.05 kernel (which doesn't support DS mode, so I can't really open .nds files), so I just renamed it to .bin or .gba on a desperate attempt but all I tried ended up on a blank screen on boot, even writing to the NOR flash (which gave a corrupt nintendo logo for some reason). 

All ROMs on the flashcart work great so I don't hink it's a problem of the cart itself. I should also note that yes, the flashcart does autopatch the ROMs but I did unpatch the file so it wouldn't give a DLDI error. I've tried other older tools but I get blank screens too (with native .bin files for the EZ4, no extension conversion or anything), some even with noise audio. 

Should I rollback the kernel? I have no idea how to make the DS mode work though, I swear I can't find any info online, I kinda know that I have to flash my DS in some way? But it sounds overcomplicated, my brother did this kinda stuff back in the day and I guess I'm unfortunately now getting into the scene. Sorry for the trouble and thanks for your work on this tool.

**Comment by AdmiralCurtiss** [comment]
Yeah this is an NDS executable and needs to be run in NDS mode. With your EZ4 (which based on some googling is a flashcard that sits in the GBA slot, right?) you need some kind of method to boot from the GBA slot in NDS mode, which from what I remember needs either:

- A flashcard that sits in the NDS slot, which may be able to chain-boot into the GBA slot while staying in NDS mode.
- Or, a hacked firmware on the DS itself, which to install you already need some way to boot custom NDS code. (look up 'flashme' if you're curious)

There were also a handful of other methods back in the day which all needed some special hardware so they're likely practically useless nowadays. You're probably best off buying a cheap NDS slot ('slot1') flashcard somewhere.

**Comment by ikorobus** [comment]
I see, thanks for the quick response, also yeah its a really really old Slot 2 cart (miniSD lmao), I actually bought a R4 just recently but I wanted to backup original DS carts with any Slot 2 tool, I'll look more stuff up and see what I can do, will also update in case I come up with something, thanks again!
